src/common/aggregate/aggregator.js

```diff
@@ -67,7 +67,17 @@ export class Aggregator {
     return hasData ? results : null
   }
 
-  #getBucket (type, name, params, customParams) {
+  /**
+   * Returns the required harvesting version for the given dataset. If ANY piece of the payload has V2 data, returns 2, otherwise 1.
+   * @param {*} aggregatorTypes
+   * @returns {1|2}
+   */
+  getRequiredVersion (aggregatorTypes) {
```

Comment:
This can just be in the EventAggregator as a sole fn; I don't see why it needs to be onion here.